src/translib/intf_app.go

```diff
@@ -131,9 +131,61 @@ func (app *IntfApp) translateReplace(d *db.DB) ([]db.WatchKeys, error) {
 func (app *IntfApp) translateDelete(d *db.DB) ([]db.WatchKeys, error) {
 	var err error
 	var keys []db.WatchKeys
-	log.Info("translateDelete:intf:path =", app.path)
+	pathInfo := app.path
 
-	err = errors.New("Not implemented")
+	log.Infof("Received Delete for path %s; vars=%v", pathInfo.Template, pathInfo.Vars)
+
+	intfObj := app.getAppRootObject()
+
+	targetUriPath, err := getYangPathFromUri(app.path.Path)
+	log.Info("uripath:=", targetUriPath)
+	log.Info("err:=", err)
+
+	if intfObj.Interface != nil && len(intfObj.Interface) > 0 {
+		log.Info("len:=", len(intfObj.Interface))
+		for ifKey, _ := range intfObj.Interface {
+			log.Info("Name:=", ifKey)
+			intf := intfObj.Interface[ifKey]
+
+			if intf.Subinterfaces == nil {
+				continue
+			}
+			subIf := intf.Subinterfaces.Subinterface[0]
+			if subIf != nil {
+				if subIf.Ipv4 != nil && subIf.Ipv4.Addresses != nil {
+					for ip, _ := range subIf.Ipv4.Addresses.Address {
+						addr := subIf.Ipv4.Addresses.Address[ip]
+						if addr != nil {
+							ipAddr := addr.Ip
+							log.Info("IPv4 address = ", *ipAddr)
+							err = app.validateIp(d, ifKey, *ipAddr)
```

Comment:
Kind of concerned about all the nested for-loops and how it might impact performance, but I guess it's unavoidable for now.

Reply:
Yes, will address those optimizations in future. But, currently its just a single instance for the primary loop. So this is not going to impact the performance as of now. Its just one DB get finally.

Reply:
But, will add a DS and keep those info, if we have get for multiple IP address.